### PR TITLE
solves 3 with recursion

### PR DESCRIPTION
**Pull request by jchancoding**

### CODE REVIEW

File: curriculum/js1/3.js [hunk 1]
```diff
@@ -5,10 +5,29 @@
  * @returns {string}
  */
 
-const solution = (a, b) => {
-  return 0;
+const solution = (x) => {
```
Comment (songz) [inline curriculum/js1/3.js]: plz remove this solution. Avoid using for loops altogether.

File: curriculum/js1/3.js [hunk 2]
```diff
@@ -5,10 +5,29 @@
...
+  return str;
 };
 
+
+//alternate solution with recursion
+solution2 = (x, string = "") => {
+  if (0 < x) {
+      x -= 1;
+      string += "hello";
+      return solution(x, string);
```
Comment (songz) [inline curriculum/js1/3.js]: start recursion with base case. 
```
if(!x) return string;
return solution(x-1, `${string}hello`);
```

File: curriculum/js1/3.js [hunk 3]
```diff
@@ -5,9 +5,15 @@
  * @returns {string}
  */
 
-const solution = (a, b) => {
-  return 0;
-};
+solution = (x, string = "") => {
+  if (!x) return string;
+  if (0 < x) {
+      string += "hello";
```
Comment (songz) [inline curriculum/js1/3.js]: You are adding hello twice.

File: curriculum/js1/3.js [hunk 4]
```diff
@@ -5,9 +5,15 @@
  * @returns {string}
  */
 
-const solution = (a, b) => {
-  return 0;
-};
+solution = (x, string = "") => {
+  if (!x) return string;
```
Comment (songz) [inline curriculum/js1/3.js]:
```
if(x <= 0) ...
return solution....
```

**Comment by jchancoding** [comment]
Repushed and it passes the test now

File: curriculum/js1/3.js [hunk 5]
```diff
@@ -5,9 +5,12 @@
  * @returns {string}
  */
 
-const solution = (a, b) => {
-  return 0;
-};
+solution = (x, string = "") => {
+  if (x <= 0) return string;
+  if (0 < x) {
```
Comment (songz) [inline curriculum/js1/3.js]: don't need this if statement.

**Comment by jchancoding** [comment]
Removed redundant if

File: curriculum/js1/3.js [hunk 6]
```diff
@@ -5,9 +5,10 @@
  * @returns {string}
  */
 
-const solution = (a, b) => {
-  return 0;
-};
+solution = (x, string = "") => {
```
Comment (songz) [inline curriculum/js1/3.js]: `const solution = (...` 
Always use const when defining variables.

**Comment by jchancoding** [comment]
Made solution into a const

**Review by songz — APPROVED** [review]

**Pull request merged**